contrib/golang/filters/http/source/go/pkg/http/type.go

```diff
@@ -182,6 +182,18 @@ func (h *requestHeaderMapImpl) Host() string {
 	return v
 }
 
+func (h *requestHeaderMapImpl) SetMethod(method string) {
```

Comment:
Why don't we provide a SetScheme like SetMethod?

Reply:
`Scheme` is the downstream protocol, may be http or https and etc. I think it might not be an good idea to override it, thoughts?

Reply:
I see. Would be better to add a comment for it.

Reply:
added an note.